python/servo/testing_commands.py

```diff
@@ -747,3 +749,51 @@ def smoketest(self, build_type: BuildType, params):
         # of panics won't cause timeouts on CI.
         return self.context.commands.dispatch('run', self.context, build_type=build_type,
                                               params=params + ['-f', 'tests/html/close-on-load.html'])
+
+    @Command('try', description='Runs try jobs by force pushing to try branch', category='devenv')
+    @CommandArgument('--remote', '-r', default="origin", help='A git remote to run the try job on')
+    @CommandArgument('try_string', default=None, nargs='...',
+                     help="A list of try strings specifying what kind of job to run.")
+    def try_command(self, remote="origin", try_string=None):
+        remote_url = subprocess.check_output(["git", "config", "--get", f"remote.{remote}.url"]).decode().strip()
+        if "github.com" not in remote_url:
+            print(f"The remote provided ({remote_url}) isn't a GitHub remote.")
+            return 1
+
+        try_string = " ".join(try_string)
+        config = servo.try_parser.Config(try_string)
+        print(f"Trying on {remote} ({remote_url}) with following configuration:")
+        print()
+        print(textwrap.indent(config.to_json(indent=2), prefix="  "))
+        print()
+
+        # The commit message is composed of both the last commit message and the try string.
+        commit_message = subprocess.check_output(["git", "show", "-s", "--format=%B"]).decode().strip()
+        commit_message = f"{commit_message} ({try_string})"
+
+        result = call(["git", "commit", "--quiet", "--allow-empty", "-m", commit_message, "-m", f"{config.to_json()}"])
+        if result != 0:
+            return result
+
+        # From here on out, we need to always clean up the commit we added to the branch.
+        try:
+            result = call(["git", "push", "--quiet", remote, "--force", "HEAD:try"])
+            if result != 0:
+                return result
+
+            # TODO: This is a pretty naive approach to turning a GitHub remote URL (either SSH or HTTPS)
+            # into a URL to the Actions page. It might be better to create this action with the `gh`
+            # tool and get the real URL.
```

Comment:
This would mean that we would depend on gh being installed and authorized and given that this is not really crucial for `mach try` I think naive approach is good enought.

Reply:
Fair point